src/fsharp/FSharp.Core/control.fs

```diff
@@ -907,9 +907,11 @@ namespace Microsoft.FSharp.Control
         let doneA           = 
             resultA()
 
+        let inline disposeIfNotNull (r:IDisposable) () = if LanguagePrimitives.IsNonNull r then r.Dispose() else ()
```

Comment:
You probably needed to use ``(r :> IDisposable).Dispose()`` since in ``usingA`` the value ``r`` has variable type (some subtype of IDisposable), but in this function it has exact type ``IDisposable``